File: Leetcode/3201 - Find the Maximum Length of Valid Subsequence I/solution.cpp
int maximumLength(vector<int> &nums)
{
    int oddCount = 0, evenCount = 0, altCount = 1, prev = nums[0];
    for (int i = 0; i < nums.size(); i++)
    {
        (nums[i] % 2) ? oddCount++ : evenCount++;
        if (i > 0 && (nums[i] % 2 != prev % 2))
        {
            altCount++;
            prev = nums[i];
        }
    }
    return max({evenCount, oddCount, altCount});
}

File: Leetcode/3201 - Find the Maximum Length of Valid Subsequence I/solution.js
var maximumLength = function (nums) {
    let oddCount = 0;
    let evenCount = 0;
    let altCount = 1;
    let prev = nums[0];

    for (let i = 0; i < nums.length; i++) {
        if (nums[i] % 2 === 0) {
            evenCount++;
        } else {
            oddCount++;
        }

        if (i > 0 && nums[i] % 2 !== prev % 2) {
            altCount++;
            prev = nums[i];
        }
    }

    return Math.max(oddCount, evenCount, altCount);
};